errors/errors.go

```diff
@@ -0,0 +1,80 @@
+package errors
```

Comment:
Errors is not a good pkg name because it is often use, for instance there is very popular pkg/errors.
So you would endup having conflicts and there is not contextual information
Let's put that to the repo root.

Reply:
> If issue states put sth to root and you want to suggest a better approach let's discuss that first.

My apologies, forgot to mention this, the linter was quite upset when I put errors.go in root directory
because I would need to import 'scylla-go-driver' and it doesn't like dashes in import, we couldn't
find fix for that so that was my temporary band-aid solution to this.